src/ApplicationManager.cpp

```
@@ -406,6 +407,22 @@ void ApplicationManager::handleSpoolDetected(const AppMessage& msg) {
         Serial.printf("ApplicationManager: Skipping LCD update for already displayed spool %s\n", msg.payload.spoolDetected.spool_id);
     }
 
+    // Apply any pending filament deductions from middleware (stored in NVS).
+    // Must run before MQTT publish so middleware sees post-deduction weight.
+#ifndef NATIVE_TEST
+    {
+        CurrentSpoolState deductState;
+        if (NFCManager::getInstance().getCurrentSpoolState(deductState) && deductState.present) {
```

Clang (14.0.6)

[note] +1, including nesting penalty of 0, nesting level increased to 1

(clang)